defos/src/defos_html5.cpp

```diff
@@ -0,0 +1,104 @@
+#include <dmsdk/sdk.h>
+#include "defos_private.h"
+
+#if defined(DM_PLATFORM_HTML5)
+
+#include <emscripten.h>
+#include "defos_private.h"
+
+static bool is_maximized = false;
+static WinRect previous_state;
+
+void js_warn(const char* msg) {
+    EM_ASM_({console.warn(UTF8ToString($0))}, msg);
+}
+
+void defos_init() {
+}
+
+void defos_final() {
+}
+
+void defos_event_handler_was_set(DefosEvent event) {
+    js_warn("Callbacks 'on_mouse_leave' and 'on_mouse_enter' don't supported in html5");
```

Comment:
These could technically be implemented with `Module.canvas.addEventListener('mouseenter')` and `Module.canvas.addEventListener('mouseleave')`

Reply:
yes, I know. 
But we have no possability to call cpp method from JS. https://github.com/defold/extension-html5/issues/1

Reply:
It should probably be safe to `ccall()` it. Setting up lua contexts is done by `defos_emit_event()` anyway. (And there's only one thread in JS anyway, so we have no such issues).

Reply:
I was think about that, but it was a deep night when i make experiments.
Let this Pull Request will be start of new platform, and we make some improvements step by step.
I will try ccall today after work, but I have to use some compilation parameters like EXPORTED_FUNCTIONS, i need to understand how to use them

Reply:
Fixed it in #48